src/Core/Environment.php

```diff
@@ -151,4 +166,92 @@ public static function getTimeLimitMax()
     {
         return static::$timeLimitMax;
     }
+
+    /**
+     * Get value of environment variable
+     *
+     * @param string $name
+     * @return mixed Value of the environment variable, or null if not set
+     */
+    public static function getenv($name)
+    {
+        switch (true) {
+            case array_key_exists($name, static::$env):
+                return static::$env[$name];
+            case array_key_exists($name, $_ENV):
+                return $_ENV[$name];
+            case array_key_exists($name, $_SERVER):
+                return $_SERVER[$name];
+            default:
+                return getenv($name);
+        }
+    }
+
+    /**
+     * Set environment variable using php.ini syntax.
+     * Acts as a process-isolated version of putenv()
+     * Note: This will be parsed via parse_ini_string() which handles quoted values
+     *
+     * @param string $string Setting to assign in KEY=VALUE or KEY="VALUE" syntax
+     */
+    public static function putenv($string)
```

Comment:
Do we need this to be our API rather than `setEnv($k, $v)`? Who would you recommend calls putenv?

Reply:
Having looked at the other code that calls it, it seems making the setEnv API public and removing putenv would be more appropriate to developers.

Reply:
I've ensure that putenv() promises to be a drop-in replacement for the core `putenv`. Mirroring the API means that users don't have to do any string-manipulation in their usercode.

Reply:
I don't see much value in that. People will still have to change their code, and if they're going to change their code we may as well give them the API they're more likely to want, which is setEnv.

Reply:
If you look at all our putenv calls we're doing string concatenation in them now, so providing `setEnv()` isn't going to force people to do string manipulation.

If we had 5 years of code built around putenv there might be a stronger argument for it, but we don't.

Reply:
ok, fair enough.

Shall I make `putEnv` and `setEnv` both public?

Reply:
Seems like a reasonable compromise.